### PR TITLE
Add a docker image for the clever tools and publish it on dockerhub

### PR DESCRIPTION
**Pull request by NotBad4U**
Work for #285

### CODE REVIEW

File: templates/dockerhub/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	mkdir -p /tmp/fakeroot/bin/ && \
+	cp /usr/local/bin/clever /tmp/fakeroot/bin/
+
+
+FROM busybox AS release
+
+LABEL version="<%= version %>" \
+	maintainer="<%= maintainer %>" \
+	description="<%= description %>" \
+	pkgver="<%= pkgver %>" \
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remote this

File: templates/dockerhub/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	sha256="<%= sha256 %>" \
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remote this

File: templates/dockerhub/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	license="<%= license %>" \
+	url="<%= url %>"
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remote this

File: templates/dockerhub/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
+FROM debian AS build
+
+RUN apt-get update && apt-get install -y \
+	libstdc++ \
+	libtool \
+	curl
+
+RUN curl --output clever-tools-latest_linux.tar.gz https://clever-tools.cellar.services.clever-cloud.com/releases/<%= version %>/clever-tools-<%= version %>_linux.tar.gz \
+	&& tar xvzf clever-tools-latest_linux.tar.gz \
+	&& cp clever-tools-latest_linux/clever /usr/local/bin
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remove latest

File: templates/dockerhub/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,41 @@
+FROM debian AS build
+
+RUN apt-get update && apt-get install -y \
+	libstdc++ \
+	libtool \
+	curl
+
+RUN curl --output clever-tools-latest_linux.tar.gz https://clever-tools.cellar.services.clever-cloud.com/releases/<%= version %>/clever-tools-<%= version %>_linux.tar.gz \
+	&& tar xvzf clever-tools-latest_linux.tar.gz \
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remove latest

File: templates/dockerhub/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,41 @@
+FROM debian AS build
+
+RUN apt-get update && apt-get install -y \
+	libstdc++ \
+	libtool \
+	curl
+
+RUN curl --output clever-tools-latest_linux.tar.gz https://clever-tools.cellar.services.clever-cloud.com/releases/<%= version %>/clever-tools-<%= version %>_linux.tar.gz \
```
Comment (hsablonniere) [inline templates/dockerhub/Dockerfile]: remove latest

File: scripts/job-publish-dockerhub.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  const underscoreVersion = version.replace(/-/g, '_');
+  const archivePath = cfg.getArchiveFilepath('linux', version);
+  const sha256 = await fs.readFile(`${archivePath}.sha256`, 'utf-8');
+
+  await cloneGitProject({ gitUrl, gitPath, git });
+  await applyOneTemplate(gitPath, templateFilepath, {
+    pkgbase,
+    pkgver: underscoreVersion,
+    version,
+    sha256,
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  const gitUrl = `ssh://git@github.com:CleverCloud/clever-tools-dockerhub.git`;
+  const version = cfg.getVersion();
+  const underscoreVersion = version.replace(/-/g, '_');
+  const archivePath = cfg.getArchiveFilepath('linux', version);
+  const sha256 = await fs.readFile(`${archivePath}.sha256`, 'utf-8');
+
+  await cloneGitProject({ gitUrl, gitPath, git });
+  await applyOneTemplate(gitPath, templateFilepath, {
+    pkgbase,
+    pkgver: underscoreVersion,
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  const pkgbase = isStableVersion ? 'clever-tools-bin' : 'clever-tools-bin-beta';
+  const gitUrl = `ssh://git@github.com:CleverCloud/clever-tools-dockerhub.git`;
+  const version = cfg.getVersion();
+  const underscoreVersion = version.replace(/-/g, '_');
+  const archivePath = cfg.getArchiveFilepath('linux', version);
+  const sha256 = await fs.readFile(`${archivePath}.sha256`, 'utf-8');
+
+  await cloneGitProject({ gitUrl, gitPath, git });
+  await applyOneTemplate(gitPath, templateFilepath, {
+    pkgbase,
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+const fs = require('fs-extra');
+const { cloneGitProject, applyOneTemplate, commitAndPush } = require('./utils');
+
+async function run () {
+
+  const templateFilepath = './templates/dockerhub/Dockerfile';
+  const gitPath = './git-docker';
+  const { git, appInfos } = cfg;
+  const isStableVersion = cfg.isStableVersion();
+  const pkgbase = isStableVersion ? 'clever-tools-bin' : 'clever-tools-bin-beta';
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  const gitUrl = `ssh://git@github.com:CleverCloud/clever-tools-dockerhub.git`;
+  const version = cfg.getVersion();
+  const underscoreVersion = version.replace(/-/g, '_');
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  const archivePath = cfg.getArchiveFilepath('linux', version);
+  const sha256 = await fs.readFile(`${archivePath}.sha256`, 'utf-8');
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+  const templateFilepath = './templates/dockerhub/Dockerfile';
+  const gitPath = './git-docker';
+  const { git, appInfos } = cfg;
+  const isStableVersion = cfg.isStableVersion();
+  const pkgbase = isStableVersion ? 'clever-tools-bin' : 'clever-tools-bin-beta';
+  const gitUrl = `ssh://git@github.com:CleverCloud/clever-tools-dockerhub.git`;
+  const version = cfg.getVersion();
+  const underscoreVersion = version.replace(/-/g, '_');
+  const archivePath = cfg.getArchiveFilepath('linux', version);
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+const cfg = require('./config');
+const fs = require('fs-extra');
+const { cloneGitProject, applyOneTemplate, commitAndPush } = require('./utils');
+
+async function run () {
+
+  const templateFilepath = './templates/dockerhub/Dockerfile';
+  const gitPath = './git-docker';
+  const { git, appInfos } = cfg;
+  const isStableVersion = cfg.isStableVersion();
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: remote this

File: scripts/job-publish-dockerhub.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const cfg = require('./config');
+const fs = require('fs-extra');
+const { cloneGitProject, applyOneTemplate, commitAndPush } = require('./utils');
+
+async function run () {
+
+  const templateFilepath = './templates/dockerhub/Dockerfile';
+  const gitPath = './git-docker';
```
Comment (hsablonniere) [inline scripts/job-publish-dockerhub.js]: rename to `git-dockerhub`

Comment (NotBad4U) [reply scripts/job-publish-dockerhub.js]: Done

File: scripts/job-publish-dockerhub.js (hunk 14 above)
Comment (NotBad4U) [reply scripts/job-publish-dockerhub.js]: Done

File: scripts/job-publish-dockerhub.js (hunk 13 above)
Comment (NotBad4U) [reply scripts/job-publish-dockerhub.js]: Done

File: scripts/job-publish-dockerhub.js (hunk 12 above)
Comment (NotBad4U) [reply scripts/job-publish-dockerhub.js]: Done

File: templates/dockerhub/Dockerfile (hunk 1 above)
Comment (NotBad4U) [reply templates/dockerhub/Dockerfile]: Done

File: templates/dockerhub/Dockerfile (hunk 2 above)
Comment (NotBad4U) [reply templates/dockerhub/Dockerfile]: Done

File: templates/dockerhub/Dockerfile (hunk 4 above)
Comment (NotBad4U) [reply templates/dockerhub/Dockerfile]: Done

File: templates/dockerhub/Dockerfile (hunk 5 above)
Comment (NotBad4U) [reply templates/dockerhub/Dockerfile]: Now we output the curl result in `clever-tools_linux`

File: scripts/job-publish-dockerhub.js (hunk 11 above)
Comment (NotBad4U) [reply scripts/job-publish-dockerhub.js]: Done

**Pull request merged**